docs/_posts/0000-05-02-oo-polymorphism.md

```diff
@@ -15,78 +11,107 @@ by <a href="https://www.flickr.com/photos/hotelrealfini/">Hotel Real Fini</a> is
 <a href="https://creativecommons.org/licenses/by/2.0/">CC BY 2.0</a>
 </p>
 
-Note:
-Again prompt for suggestions and discuss.  
-Interfaces not really required in JS, but there are mechanisms.  
-
 --
 
-### Example - Lights on a car
-
-+ A car has many lights
-    + Headlights
-    + Side-lights
-    + Indicators
-    + Fog-lights
-+ All lights can be turned on and off by controls on the dashboard
+### Existing examples
 
---
+* We already have examples of polymorphism in our code
 
 ```java
-abstract class Light {
-    void turnOn(){}
-    void turnOff(){}
-}
-class Headlight extends Light {}
-class Indicator extends Light {}
-class FogLight extends Light {}
-class Car {
-    Light[] lights = {
-        new Headlight(), new Indicator(), new FogLight()
-    };
-    Car(){
-        lights.forEach(l -> l.turnOn());
-    }
-}
+  Deck deck = new PlayingCardDeck();
+  deck.shuffle();
+  deck.deal();
+  
+  Card card = deck.deal();
+  card.snap(previousCard);
 ```
 
 --
 
+### How do we use polymorphism?
+
+* We achieve polymorphism by using inheritance / interfaces to express relationships between classes
++ Why is it useful?
+    + Simplifies our code
+    + Removes `if` checks
+
+
+
+Note:
+Again prompt for suggestions and discuss.  
+Interfaces not really required in JS, but there are mechanisms.  
+
+--
+
+
 ### Exercise (30mins)
 
-* Make a game of _**Super-Snap!**_
+* Make a game of _**Super-Snap**_ by creating a new MixedDeck class
 * The deck consists of a full set of playing cards and a full set of animal cards
 * The rules for calling _Snap!_ are the same as before
-    + a playing card will only match a playing card of the same value
-    + an animal card will never match a normal playing card
+    + a playing card will only match a playing card of the same face value
+    + an animal card will only match another animal card of the same creature
+* Make your Snap class use the new MixedDeck
+
+--
+
+### Where's the polymorphism?
+
+         ┌───────────────────────────────────────────────┐
+         │  MixedDeck                                    │
+         │                                               │
+         │  Card[] cards = { "The ace of spades",        │
+         │                   "Gorilla",                  │
+         │                   "Cheeta",                   │
+         │                   "The seven of diamonds" };  │
+         │                                               │
+         │                                               │
+         └───────────────────────────────────────────────┘
```

Comment:
For consistency, I’d use triple backticks here to describe the fenced code block.
````suggestion
```
 ┌───────────────────────────────────────────────┐
 │  MixedDeck                                    │
 │                                               │
 │  Card[] cards = { "The ace of spades",        │
 │                   "Gorilla",                  │
 │                   "Cheeta",                   │
 │                   "The seven of diamonds" };  │
 │                                               │
 │                                               │
 └───────────────────────────────────────────────┘
 ```
````